client/asset/eth/eth.go

```diff
@@ -697,8 +698,63 @@ func (eth *ExchangeWallet) SignMessage(coin asset.Coin, msg dex.Bytes) (pubkeys,
 // AuditContract retrieves information about a swap contract on the
 // blockchain. This would be used to verify the counter-party's contract
 // during a swap.
-func (*ExchangeWallet) AuditContract(coinID, contract, txData dex.Bytes, rebroadcast bool) (*asset.AuditInfo, error) {
-	return nil, asset.ErrNotImplemented
+func (eth *ExchangeWallet) AuditContract(coinID, contract, txData dex.Bytes, rebroadcast bool) (*asset.AuditInfo, error) {
+	tx := new(types.Transaction)
+	err := tx.UnmarshalBinary(txData)
+	if err != nil {
+		return nil, fmt.Errorf("AuditContract: failed to unmarshal transaction: %w", err)
+	}
+
+	txHash := tx.Hash()
+	if !bytes.Equal(coinID, txHash[:]) {
+		return nil, fmt.Errorf("AuditContract: coin id != txHash - coin id: %x, txHash: %x", coinID, tx.Hash())
+	}
+
+	initiations, err := dexeth.ParseInitiateData(tx.Data())
```

Comment:
OK, so with the push toward versioned contracts, we'd have to do the same encoding scheme with (ver|secretHash) for the `contract` input, decode the version, and then use a version-aware `dexeth.ParseInitiateData`, returning a new `SwapInitiation` type that is translated internally from each version's `ETHSwapInitiation`.

This would mean that either `msgjson.Audit.Contract` from the server is gonna need that encoding, or when we call `AuditContract`, we prepend the version we're expecting based on the server config.  I'm not yet clear on which way we want to go with this.

Reply:
Oh, sorry it would just be in `OrderMetaData.{FromVersion,ToVersion}`.  Just getting used to the new scheme from 1301...

Reply:
My vote is to prepend the version.

Reply:
I think we can just use `*asset.Contract` as the common type.

Reply:
Sorry to spam the comments here, but I just realized that the version is already prepended, since the counter-party's `swapReceipt` encodes it that way.

Reply:
I'm working on some server updates for contract versioning, but yeah, @buck54321 is right that the `msgjson.Audit.Contract` passed to `AuditContract` already has it.

Reply:
Things are changing quickly lately!

Reply:
They certainly are.  Last night I started on updates for server to handle the version in what I'm calling contract data from the swap receipt and thus in the `msgjson.Init.Contract` and everywhere downstream:  https://github.com/decred/dcrdex/pull/1320
I will have that follow https://github.com/decred/dcrdex/pull/1313, which will follow https://github.com/decred/dcrdex/pull/1307